executor/metric_reader.go

```diff
@@ -52,29 +55,44 @@ func (e *MetricRetriever) retrieve(ctx context.Context, sctx sessionctx.Context)
 		return nil, err
 	}
 	e.tblDef = tblDef
-	// TODO: Get query range from plan instead of use default range.
-	queryRange := e.getDefaultQueryRange(sctx)
-	queryValue, err := e.queryMetric(ctx, sctx, queryRange)
+	queryRange := e.getQueryRange(sctx)
+	quantiles, err := e.extractor.GetQuantiles()
 	if err != nil {
 		return nil, err
 	}
+	rows := make([][]types.Datum, 0)
+	for _, quantile := range quantiles {
+		if quantile == 0 {
+			quantile = tblDef.Quantile
+		}
+		queryValue, err := e.queryMetric(ctx, sctx, queryRange, quantile)
+		if err != nil {
+			return nil, err
+		}
 
-	fullRows = e.genRows(queryValue, queryRange)
-	if len(e.outputCols) == len(e.table.Columns) {
-		return
-	}
-	rows := make([][]types.Datum, len(fullRows))
-	for i, fullRow := range fullRows {
-		row := make([]types.Datum, len(e.outputCols))
-		for j, col := range e.outputCols {
-			row[j] = fullRow[col.Offset]
+		fullRows = e.genRows(queryValue, queryRange, quantile)
+		if len(e.outputCols) == len(e.table.Columns) {
+			rows = append(rows, fullRows...)
+			continue
+		}
+		for _, fullRow := range fullRows {
+			row := make([]types.Datum, len(e.outputCols))
+			for j, col := range e.outputCols {
```

Comment:
Is it necessary? the column prune will not apply to `PhyicalMemTable`.

Reply:
Great, done.